script.rb

```diff
@@ -0,0 +1,152 @@
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return false unless yield i }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+  #-------------------my_any?-----------------------
+
+  def my_any?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return true if !i[args].nil? || i[args] == 1 } # if Regexp
+      else
+        my_each { |i| return true if i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return true if i.is_a?(args) }
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return true if yield i }
+    else
+      my_each { |i| return false unless i.nil? || i == false }
+    end
+    false
+  end
+
+  #-------------------my_none?-----------------------
+  def my_none?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return false unless i[args].nil? || i[args] != 1 } # if Regexp
+      else
+        my_each { |i| return false unless i != args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return false if i.is_a?(args) }
+
+    elsif args.nil? && block_given?
+      my_each { |i| return false if yield i }
+    else
+      my_each { |i| return false unless i.nil? || i != false }
+    end
+    true
+  end
+
+  #-------------------my_count-----------------------
+  def my_count(args = nil)
+    count = 0
+    if !args.nil?
+      my_each { |x| count += 1 if x == args }
+    elsif args.nil? && block_given?
+      my_each { |x| count += 1 if yield x }
+    else my_each { count += 1 }
+    end
+    count
+  end
+
+  #-------------------my_map-----------------------
+  def my_map(args = nil)
+    return to_enum(:my_map) if args.nil? && !block_given?
+
+    mapped = []
+    if args.nil?
+      my_each_with_index { |i, j| mapped[j] = yield i }
+    else
+      my_each_with_index { |i, j| mapped[j] = args.call(i) }
+    end
+    mapped
+  end
+  #-------------------my_inject-----------------------
+
+  def my_inject(arg1 = nil, arg2 = nil)
+    if block_given?
+      acc = arg1
+      my_each { |item| acc = acc.nil? ? item : yield(acc, item) }
+      acc
+    elsif arg1.is_a?(Symbol) || arg1.is_a?(String)
+      acc = nil
+      my_each { |item| acc = acc.nil? ? item : acc.send(arg1, item) }
+      acc
+    elsif arg2.is_a?(Symbol) || arg2.is_a?(String)
+      acc = arg1
+      my_each { |item| acc = acc.nil? ? item : acc.send(arg2, item) }
+      acc
+    else
+      raise LocalJumpError, 'No block Given or Empty Argument' unless !arg1.nil? && !arg2.nil? && !block_given?
+    end
+  end
+
+  #-------------------multiply_els-----------------------
+  def multiply_els(arr)
+    raise TypeError, 'No Array Given' if arg.nil? || !arg.is_a?(Array)
+
+    arr.my_inject { |result, element| result * element }
+  end
```

Comment:
should have working `multiply_els` that uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- [ ] accepts an array as an argument and multiplies all the elements of the array together using #my_inject

```ruby
multiply_els([5, 4, 20]) => 400
```
- This is the error that is being retrieved.
![image](https://user-images.githubusercontent.com/52209067/93889850-eed61680-fca6-11ea-9d03-a15a53fbf132.png)

- [ ] multipy_els is not an enumerable method so let's take it out of the Enumerable module and put it in a different one